nano/lib/blocks.cpp

```diff
@@ -161,45 +208,118 @@ bool nano::block::has_sideband () const
 	return sideband_m.is_initialized ();
 }
 
-nano::account const & nano::block::representative () const
+std::optional<nano::account> nano::block::representative_field () const
 {
-	static nano::account representative{};
-	return representative;
+	return std::nullopt;
 }
 
-nano::block_hash const & nano::block::source () const
+std::optional<nano::block_hash> nano::block::source_field () const
 {
-	static nano::block_hash source{ 0 };
-	return source;
+	return std::nullopt;
 }
 
-nano::account const & nano::block::destination () const
+std::optional<nano::account> nano::block::destination_field () const
 {
-	static nano::account destination{};
-	return destination;
+	return std::nullopt;
 }
 
-nano::link const & nano::block::link () const
+std::optional<nano::link> nano::block::link_field () const
 {
-	static nano::link link{ 0 };
-	return link;
+	return std::nullopt;
 }
 
-nano::account const & nano::block::account () const
+nano::account nano::block::account () const
 {
-	static nano::account account{};
-	return account;
+	release_assert (has_sideband ());
+	switch (type ())
+	{
+		case block_type::open:
+		case block_type::state:
+			return account_field ().value ();
+		case block_type::change:
+		case block_type::send:
+		case block_type::receive:
+			return sideband ().account;
+		default:
+			release_assert (false);
+	}
+}
+
+nano::amount nano::block::balance () const
+{
+	release_assert (has_sideband ());
+	switch (type ())
+	{
+		case nano::block_type::open:
+		case nano::block_type::receive:
+		case nano::block_type::change:
+			return sideband ().balance;
+		case nano::block_type::send:
+		case nano::block_type::state:
+			return balance_field ().value ();
+		default:
+			release_assert (false);
+	}
+}
+
+nano::account nano::block::destination () const
+{
+	release_assert (has_sideband ());
+	switch (type ())
+	{
+		case nano::block_type::send:
+			return destination_field ().value ();
+		case nano::block_type::state:
+			release_assert (sideband ().details.is_send);
+			return link_field ().value ().as_account ();
+		default:
+			release_assert (false);
```

Comment:
There is some inconsistency in the way unexpected behavior is handled, `.value ()` will throw an exception while `release_assert` will terminate the program. Maybe those accessors should be marked with `noexcept` to terminate instantly if they find the program in undefined state.

Reply:
Yea I can make that change. The intent with the value () calls is to terminate, at least in these inner functions.